### ISSUE TITLE
EnC doesn't detect local function closure to lambda closure udpate

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#67323]
If a new lambda captures a variable that is alredy captured by a local function then the closure type is converted from struct local function closure to a lambda display class.

Similarly, when a new conversion from local function group to a delegate is added the closure type also changes.

Both should be reported as rude edits during capture analysis.

```C#
class C
{
    void F()
    {
        int t = 0;
        int L() => t;
        
        // var f = () => t;   /* case 1 */
        // Delegate d = L;    /* case 2 */
    }
}
```

Related: https://github.com/dotnet/roslyn/issues/66680
